### PR TITLE
Fix #10286: Unable to delete network rules

### LINKED ISSUE #10286: Cannot remove VNET subnet for KeyVault when subnet ID contains uppercase letters
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**
When run 
az keyvault network-rule remove --subnet "<my-subnet-id>" --resource-group "<my-resource-group>" --name "<my-key-vault-name>"
It fails to remove the subnet without error when "<my-subnet-id>" contains upper-case letters.

**To Reproduce**
Steps to reproduce the behavior.

The following command will not remove subnet because text “/Microsoft.Network/virtualNetworks/”:
az keyvault network-rule remove --subnet "/subscriptions/11111111-aa7f-4b05-86b7-61be294b8be0/resourcegroups/foo-vnet-res-group/providers/Microsoft.Network/virtualNetworks/foo-vnet-test-02/subnets/foo-test02" --resource-group "foo-vnet-res-group" --name "foo-vnet-test-33"

The following command does work because subnet-id contains all lower-case letters:
az keyvault network-rule remove --subnet "/subscriptions/11111111-aa7f-4b05-86b7-61be294b8be0/resourcegroups/foo-vnet-res-group/providers/microsoft.network/virtualnetworks/foo-vnet-test-02/subnets/foo-test02" --resource-group "foo-vnet-res-group" --name "foo-vnet-test-33"

**Expected behavior**
The subnet should be removed, e.g. the command “az keyvault network-rule list” should not include removed subnet ID.

**Environment summary**
Install Method (e.g. pip, interactive script, apt-get, Docker, MSI, edge build) / CLI version (`az --version`) / OS version / Shell Type (e.g. bash, cmd.exe, Bash on Windows)  
Latest CLI in Window 10

**Additional context**

### PR DESCRIPTION
**Pull request by jiasli**
Fix #10286: Unable to delete network rules due to subnet ID getting converted to lowercase by keyvault service.

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/keyvault/custom.py [hunk 1]
```diff
@@ -489,7 +489,15 @@ def add_network_rule(cmd, client, resource_group_name, vault_name, ip_address=No
 
     if subnet:
         rules.virtual_network_rules = rules.virtual_network_rules or []
-        rules.virtual_network_rules.append(VirtualNetworkRule(id=subnet))
+
+        # if the rule already exists, don't add again
+        modified = True
```
Comment (yugangw-msft) [inline src/azure-cli/azure/cli/command_modules/keyvault/custom.py]: suggest name as `to_modify`

File: src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py [hunk 2]
```diff
@@ -888,5 +888,46 @@ def test_keyvault_storage_account(self, resource_group):
...
+        test.cmd('network vnet create -g {rg} -n {vnet} -l {loc} ')
+        test.cmd('network vnet subnet create -g {rg} --vnet-name {vnet} --name {subnet} '
+                 '--address-prefixes 10.0.0.0/21 --service-endpoints Microsoft.KeyVault')
+        return test.cmd('network vnet subnet show -g {rg} --vnet-name {vnet} --name {subnet}')
+
+    @ResourceGroupPreparer(name_prefix='cli_test_keyvault_network_rule')
+    def test_keyvault_network_rule(self, resource_group):
+
+        self.kwargs.update({
+            'kv': self.create_random_name('cli-test-keyvault-', 24),
```
Comment (yugangw-msft) [inline src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py]: like we discussed on race condition unique to key-vault tests, consider using a unique prefix

File: src/azure-cli/azure/cli/command_modules/keyvault/custom.py (hunk 1 above)
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/keyvault/custom.py]: Thanks. Pushed new commit.

File: src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py (hunk 2 above)
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py]: All certs have expired. Need more time to fix them. Let's fix firewall issue first.

**Pull request merged**